lang/mlton-bootstrap/Portfile

```diff
@@ -0,0 +1,138 @@
+    if {${os.major} > 18} {
+        version     20210117
+        revision    0
+        # What can we do…
+        distfiles   mlton-${version}-1.amd64-darwin-19.6.gmp-homebrew.tgz
+        checksums   rmd160  fbceabd8dac11cff7f839f953391596a3f6222ed \
+                    sha256  4f50973e9ec47ffb3e7ce5ecd14996e3894779eec788ce464e6c9a903719cf28 \
+                    size    18056959
+        patchfiles-append \
+                    patch-x86_64.diff
+    } elseif {${os.major} > 16} {
+        version     20201002
+        revision    0
+        distfiles   mlton-${version}-1.amd64-darwin-17.7.gmp-homebrew.tgz
+        checksums   rmd160  be93f0616386602f68ba33a981f4116545137465 \
+                    sha256  0abd8ddb8dd51d078b4b7f400f488d069d53017333c78090d3b6a732b49f81ac \
+                    size    17824913
+        patchfiles-append \
+                    patch-x86_64.diff
+    } else {
+        version     20180207
+        revision    0
+        distfiles   mlton-${version}-1.amd64-darwin.gmp-homebrew.tgz
+        checksums   rmd160  3e5374ed9a12906409c74ba3cc790222b4d877ea \
+                    sha256  c164c1f3a3bbce0a0008708984202ac8f7d4dc1a3d2aa83a4faeb41e6d68bf2f \
+                    size    18874573
+        patchfiles-append \
+                    patch-${version}.diff
+    }
+} elseif {${configure.build_arch} eq "arm64"} {
+        version     20210117
+        revision    0
+        distfiles   mlton-${version}-1.arm64-darwin-21.6-gmp-dynamic.tgz
+        checksums   rmd160  64fa80a6ac321f7d36dafe41a3ef21df95bacf9a \
+                    sha256  d8acf4bb925abd2aa509c318c59c3c5ccd10813b098aa99292498878e7da3bfa \
+                    size    8397794
+        patchfiles-append \
+                    patch-aarch64.diff
+}
+
+if {${configure.build_arch} eq "arm64"} {
+    master_sites    https://projects.laas.fr/tina/software/
+} else {
+    master_sites    sourceforge:project/mlton/mlton/${version}
+}
+
+extract.rename      yes
+
+depends_lib         port:gmp
+
+universal_variant   no
+
+use_configure       no
+
+post-patch {
+    reinplace "s|@CC@|${configure.cc}|" ${worksrcpath}/bin/mlton
+    reinplace "s|@MLTON@|${prefix}/libexec/mlton-bootstrap/lib/mlton|" ${worksrcpath}/bin/mlton
+    reinplace "s|@PREFIX@|${prefix}|g" ${worksrcpath}/bin/mlton
+}
+
+# Not required here, but needed to build the current MLton, and we want to use the same compiler:
+compiler.c_standard 2011
+compiler.blacklist-append {clang < 900}
+
+# While newer versions got makefiles in a bundle, they are optional: https://github.com/MLton/mlton/issues/501
+build {}
+
+destroot {
+    xinstall -m 755 -d ${destroot}${prefix}/libexec/${name}
+
+    foreach dir [glob ${worksrcpath}/*] {
+        copy $dir ${destroot}${prefix}/libexec/${name}
+    }
+}
+
+post-destroot {
+    # These are dylibs which original binaries were built against:
+    if {${configure.build_arch} eq "ppc"} {
+        set old_gmp /opt/local/lib/libgmp.3.dylib
```

Comment:
The lint warnings are already complaining about this, but do no hardcode `/opt/local` - that's not going to be the case for every MacPorts installation.  Use `${prefix}` instead.

Reply:
It is a bootstrap port using pre-built binaries. `old_gmp` is the path which was used in 2007 or w/e by someone who built them. To change it to a correct path used presently by a Macports user, old path must be hardcoded, it is just a given value.

Reply:
OK, noted.